### PR TITLE
Ensure existing DHCP server is in ACTIVE state

### PR DESCRIPTION
**Pull request by dharaneeshvrd**
**What this PR does / why we need it**:
Fixes https://issues.redhat.com/browse/MULTIARCH-2961

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Review by mkumatag — APPROVED** [review]
/lgtm

File: cmd/infra/powervs/create.go [hunk 1]
```diff
@@ -1148,6 +1159,28 @@ func (infra *Infra) setupPowerVSDHCP(ctx context.Context, options *CreateInfraOp
 	return nil
 }
 
+// monitorDhcpStatus monitor DHCP status to reach either ACTIVE or ERROR status
```
Comment (enxebre) [inline cmd/infra/powervs/create.go:1162]: nit: should be monitorDHCPStatus

File: cmd/infra/powervs/create.go [hunk 2]
```diff
@@ -1148,6 +1159,28 @@ func (infra *Infra) setupPowerVSDHCP(ctx context.Context, options *CreateInfraOp
...
+	}
+
+	if dhcpServer != nil {
+		log(infraID).Info("Waiting for DHCPServer to up", "id", *dhcpServer.ID, "status", *dhcpServer.Status)
+		if *dhcpServer.Status == dhcpServiceActiveState {
+			return dhcpServer, true, nil
+		}
+
+		if *dhcpServer.Status == dhcpServiceErrorState {
+			return dhcpServer, true, fmt.Errorf("dhcp service is in error state")
```
Comment (enxebre) [inline cmd/infra/powervs/create.go:1177]: Why/when this function return true/false? is not clear by the name.

File: cmd/infra/powervs/create.go (hunk 1 above)
Comment (dharaneeshvrd) [reply cmd/infra/powervs/create.go]: done.

File: cmd/infra/powervs/create.go (hunk 2 above)
Comment (dharaneeshvrd) [reply cmd/infra/powervs/create.go:1178]: Updated the comment explaining the return values. ptal!

Comment (enxebre) [reply cmd/infra/powervs/create.go:1178]: >// returns an instance of DHCPServerDetail for further processing and true if reaches either ACTIVE or ERROR status

I see, thanks.  Then you use isReady to gather the value which makes me think the return value is evaluating if is ACTIVE or not, but that's not what's happening.

Should the function return true when isActive and so rename the variable?

Should we I'd rather break this down into two functions getDHCPServer() and isDHCPServerActive()?

**Pull request merged**